docs/gfm-alerts-to-admonitions/gfm_alerts_to_admonitions/main.py

```diff
@@ -0,0 +1,51 @@
+import re
+
+import markdown_it
+
+_ALERT_PATTERN = re.compile(
+    r"^\[\!(TIP|NOTE|IMPORTANT|WARNING|CAUTION)\]\s*", re.IGNORECASE
+)
+
+
+def gfm_alert_to_admonition(md: markdown_it.MarkdownIt):
+    def gfm_alert_to_adm(state: markdown_it.rules_core.StateCore):
+        tokens = state.tokens
+        for i, token in enumerate(tokens):
+            if token.type == "blockquote_open":
+                # Store the start of the blockquote (as a token, not a char)
+                open = tokens[i]
+                start_index = i
+
+                # Search for the end
+                while tokens[i].type != "blockquote_close" and i < len(tokens):
+                    i += 1
+
+                # Store the end of the blockquote
+                close = tokens[i]
+                end_index = i
+
+                # Find the first `inline` token in the blockquote
+                first_content = next(
+                    (
+                        t
+                        for t in tokens[start_index : end_index + 1]
+                        if t.type == "inline"
+                    ),
+                    None,
+                )
+                if first_content is None:
+                    continue
+
+                if (m := _ALERT_PATTERN.match(first_content.content)) is None:
+                    continue
+
+                # Remove ' [!TIP]' from the token
+                first_content.content = first_content.content[len(m[0]) :]
```

Comment:
Looks like you tried `m.endpos` but reverted. Did it not do what I thought it would?

Reply:
Yeah, I responded in the original suggestion with what it really does

Reply:
(I was glad to have the tests, as they caught the issue immediately)